metabase/metabase
# Add actions tab to model detail page

Epic #27581

Adds an "Actions" tab to the model detail page. The new section lets people to:

* browse actions associated with a model
* create basic actions (generic create, update and delete that are tracking schema changes automatically)
* create query actions (custom SQL with parameters for stuff like `UPDATE WHERE id = {{ id }}`)

Out of scope for now: archiving actions, displaying more info (blocked by BE for now)

### To Verify

1. Sign in as admin
2. `PUT api/setting/experimental-enable-actions` with `{ value: true }` to enable actions globally (will be removed later)
3. Go to `/admin/databases/:id` and enable actions for your database in the bottom right part of the page
4. Open `/model/:id/detail` and click the "Actions" tab
5. Make sure you can see an empty state with a "New action" button and CTA to create basic (implicit) actions
6. Click on "Create basic actions", and ensure you can see Create, Update and Delete actions appeared
7. Click "New action", write a custom SQL query, and save
8. Ensure you can see your action in the list
9. Click the pencil icon to edit the action, ensure you can actually do it

### Demo

https://user-images.githubusercontent.com/17258145/212748394-4dddcdbf-b114-4c56-b660-35905c940bd6.mp4


- - -
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/metabase/metabase/27657)
